### PR TITLE
Faster quantiles

### PR DESCRIPTION
**Pull request by Felixmil**
Previously, the `quantile()` function was called 3 times (once for each quantile).

This function can be slow on large dataset because it requires sorting the data.

With this PR, we use the function `quantile()`  just once and then save 2 sorting steps.

Other functions are used so the output data structure remains the same.

### CODE REVIEW

**Comment by Felixmil** [comment]
@PavelBal, This is one of the reasons the `plotPopulationTimeProfile()` calls were extremely slow in our recent project.

**Comment by Felixmil** [comment]
## Performances 🚀

With the first proposed solution, the time for computing quantiles is reduced by ~60% on average for 10 K rows.
However, the allocated memory increased by 2.8 times. 

With the latest solution which uses `{data.table}` we achieve the following performances:
- ~ -63% time compared to original implementation (-9% compared to first solution)
- ~ -68% of used memory compared to original and (-89% compared to first solution)

The downside is that OSPSR would now depend on  `{data.table}` 

## Details 🔎

|expression     |      min|   median|   itr/sec| mem_alloc|   gc/sec| n_itr| n_gc| total_time|
|:--------------|--------:|--------:|---------:|---------:|--------:|-----:|----:|----------:|
|original_q(d)  |    1.31s|    1.43s| 0.6937766|    2.36MB| 18.78053|   100| 2707|       2.4m|
|one_q(d)       | 496.94ms| 584.86ms| 1.6950686|    6.62MB| 15.69634|   100|  926|        59s|
|datatable_q(d) | 488.36ms| 533.79ms| 1.8154798|   763.2KB| 13.43455|   100|  740|      55.1s|


## Quality ✅

```
══ Results ═══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
Duration: 247.1 s


[ FAIL 0 | WARN 10 |  PASS 1235 ]
```

## Reproducibility 📜
```r
library(tidyverse)
library(data.table)


nrows <- 1e4 #10 000 rows
d <- data.frame("a"=sample(1:nrows/3, nrows, TRUE), "b"=rnorm(nrows), "c" = sample(1:2, nrows,replace = T))
quantiles <- c(0.05, 0.5, 0.95)

original_q <- function(df){
  df %>%
    group_by(c, a) %>%
    summarise(l = quantile(b, quantiles[1]),
              m = quantile(b, quantiles[2]),
              h = quantile(b, quantiles[3]),
              .groups = "drop")
}


one_q <- function(df){
  quantileNames <- c("l", "m", "h")
  
  df %>%
    group_by(c, a) %>%
    summarise(quantiles = list(quantile(b, quantiles)),
              names = list(quantileNames),
              .groups = "drop") %>%
    unnest(c(quantiles, names)) %>% # expand lists
    pivot_wider(names_from = names, values_from = quantiles)
}


datatable_q <- function(df){
  quantileNames <- c("l", "m", "h")
  
  df = data.table::as.data.table(df)
  
  df[,setNames(as.list(quantile(b, quantiles)), quantileNames), by = .(c, a)]
}

result <- bench::mark(original_q(d), one_q(d), datatable_q(d),
                      iterations = 100, 
                      check = F)
```

**Comment by Yuri05** [comment]
> is `data.table` part of valid R?

yes

**Review by PavelBal — DISMISSED** [review]

File: DESCRIPTION [hunk 1]
```diff
@@ -29,7 +29,8 @@ Imports:
     tidyr,
     ggplot2,
     rlang,
-    tlf (>= 1.4.0)
+    tlf (>= 1.4.0),
+    data.table
```
[the remark this answers is not in the document]
Comment (PavelBal) [reply DESCRIPTION:33]: @Felixmil please also add it to the list of installed packages in https://github.com/esqLABS/esqlabsR/blob/develop/inst/extdata/install-packages.R

[the remark this answers is not in the document]
Comment (Felixmil) [reply DESCRIPTION:33]: @Yuri05 Done

**Review by PavelBal — APPROVED** [review]

**Pull request merged**